config/formats.ts

```diff
@@ -1629,6 +1629,31 @@ export const Formats: FormatList = [
 		searchShow: false,
 		ruleset: ['Obtainable', 'HP Percentage Mod', 'Cancel Mod', 'Desync Clause Mod'],
 	},
+	{
+		name: "[Gen 1] Hackmons Cup",
+		desc: `Randomized teams of level-balanced Pok&eacute;mon with absolutely any moves, types, and stats.`,
+
+		mod: 'gen1',
+		team: 'randomHC',
+		searchShow: false,
+		ruleset: ['HP Percentage Mod', 'Cancel Mod', 'Desync Clause Mod', 'Sleep Clause Mod', 'Freeze Clause Mod', 'Team Preview'],
+		onBegin() {
+			for (const pokemon of this.getAllPokemon()) {
+				this.add(`c|\u2606${pokemon.side.name}|${pokemon.name}'s types: ${(pokemon.set as any).hc.types.join('/')}`);
```

Comment:
I think it's a bit weird to make the messages come from the player - during testing you managed to get it to just show up as the small text in battles like the rules. If you're going to use `c|` I'd rather it come from `&Staff` or `&Battle` rather than the plater.